File: src/main/java/com/amigos/hacer/porhacer/models/Task.java
package com.amigos.hacer.porhacer.models;


import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import java.util.Set;

@Data
@Entity
@Table(name = "tasks")
public class Task {

    @Id
    @GeneratedValue
    private Long id;
    private String name;
    private String description;
    private TaskStatus status;

    //groups that this task belongs to
    @ManyToMany(mappedBy = "tasks")
    private Set<Group> groups;

    @ManyToOne
    @JoinColumn(name="group_id", nullable=false)
    private Group group;

}

File: src/main/kotlin/com/amigos/amigos/models/Task.kt
package com.amigos.amigos.models

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id


@Entity
class Task(var task: String, var isCompleted: Boolean) {
    constructor() : this("", false)

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY) // this is the primary key which will be auto generated
    var id: Long? = null



}